src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -84,6 +84,16 @@ public void addMessage(Message message) {
 
     return messagesInConversation;
   }
+  public List<Message> getMessagesByUser(UUID id){
+
+    List<Message> messagesByUser = new ArrayList<>();
+    for(Message message : messages){
+      if(message.getAuthorId().equals(id)){
+        messagesByUser.add(message);
+      }
+    }
+    return messagesByUser;
+  }
```

Comment:
In the latest version of the project, we have already a method `getMessagesFromUser(User user)`. Maybe we should reuse it. However, I think your implementation is more efficient.
@khankuan what do you think?

Reply:
I agree this implementation is better. Maybe @Dnae-ferguson can change `getMessagesFromUser` in this file to reuse `getMessagesByUser` by calling `getMessagesFromUser`.

It might be clearer to name it `getMessagesFromUserId` as well.